backend/routes/officers.py

```diff
@@ -233,21 +234,46 @@ def get_all_officers():
         if key in and_clauses and args[key]:
             cypher_query += "\n" + and_clauses[key]
 
-    cypher_query += "\nRETURN o"
+    count_results, _ = db.cypher_query(
+        cypher_query + "\nReturn count(*) as c", args)
+
+    row_count = count_results[0][0] if count_results else 0
+    logging.warning(f'Total results found: {row_count}')
+
+    if row_count == 0:
+        return jsonify({"message": "No results found matching the query"}), 200
+
+    if row_count < (q_page - 1) * q_per_page:
+        return jsonify({"message": "Page number exceeds total results"}), 400
+
+    skip = (q_page - 1) * q_per_page
+    limit = q_per_page
+
+    # cypher_query += "\nRETURN o"
+    cypher_query += f"\nRETURN o skip {skip} limit {limit}"
     logging.warning(f'cypher query is {cypher_query}')
 
+    # run the full query
     results, meta = db.cypher_query(cypher_query, args)
     logging.warning(f'cypher results are {results}')
     logging.warning(f'number of results is {len(results)}')
-    all_officers = [Officer.inflate(row[0]) for row in results]
 
+    all_officers = [Officer.inflate(row[0]) for row in results]
     logging.warning(f'API: number of officers is {len(all_officers)}')
+
     if not all_officers:
         logging.warning("No officers found with the given filters")
         return ordered_jsonify([]), 200
 
-    results = paginate_results(list(all_officers), q_page, q_per_page)
-    return ordered_jsonify(results), 200
+    page = [item.to_dict() for item in all_officers]
+
+    # results = paginate_results(list(all_officers), q_page, q_per_page)
+    response = add_pagination_wrapper(
+        page_data=page, total=row_count,
+        page_number=q_page, per_page=q_per_page)
+
+    logging.warning(f'API: response is {response}')
+    return jsonify(response), 200
```

Comment:
Here you want to import and use [`ordered_jsonify`](https://github.com/codeforboston/police-data-trust/blob/c4f433a5692ad6efad5365ac2813b8d8e66788aa/backend/schemas.py#L102). It'll make sure that the properties are serialized and hidden properties (like citations) get removed.